readyapi4j-core/src/main/java/com/smartbear/readyapi4j/execution/DataExtractors.java

```diff
@@ -0,0 +1,59 @@
+package com.smartbear.readyapi4j.execution;
+
+import com.smartbear.readyapi.client.model.ProjectResultReport;
+import com.smartbear.readyapi.client.model.TestCaseResultReport;
+import com.smartbear.readyapi.client.model.TestSuiteResultReport;
+import com.smartbear.readyapi4j.extractor.ExtractorData;
+import com.smartbear.readyapi4j.extractor.ExtractorOperator;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Utility class to execute the data extractions after recipe execution.
+ */
+public class DataExtractors {
+    public static void runDataExtractors(ProjectResultReport executionStatus, List<ExtractorData> extractorDataList) {
+        List<String> extractorDataIdList = extractorDataList
+                .stream()
+                .map(ExtractorData::getExtractorDataId)
+                .collect(Collectors.toList());
+        TestCaseResultReport resultReport = executionStatus
+                .getTestSuiteResultReports()
+                .stream()
+                .map(TestSuiteResultReport::getTestCaseResultReports)
+                .flatMap(Collection::stream)
+                .filter(testCaseResultReport ->
+                        extractorDataIdList.contains(testCaseResultReport.getProperties().get(ExtractorData.EXTRACTOR_DATA_KEY)))
+                .findAny()
+                .orElse(null);
```

Comment:
Why not let the type of this variable just be Optional<TestCaseResultReport> instead?